### DISCUSSION TITLE
Reporting Entity Attributes in both Table_Of_Contents and INN file

### DISCUSSION THREAD

**Discussion by angelamarie0975** [CMSgov/price-transparency-guide#370] [Q&A]
The Reporting Entity attributes are on both the TOC and INN files which seems redundant in both files and will cause issuers to have to duplicate INN files, which was one of the main issues that the TOC file addressed.

We will have 2 separate TOC files for most of our plans as they have both ASO plans and standard plans per product line - the ASO plans will have a Reporting_Entity_Type = "Third Party Administrator" and the Standard plans will have a Reporting_Entity_Type = "Health Insurance Issuer". Regardless of ASO or Standard plans - they can use the same network of providers, which means both TOC files will have reference to the same INN files - this is expected and no real issues with this. **However, ** since the INN file has the reporting entity attributes as well, this means that there will need to be 2 separate INN files because based on the current schema, we must include the reporting entity attributes which are going to be different. The Reporting Entity Name will remain the same, but the type will be different, and this is going to cause us to have to create 2 files per individual network instead of just the 1. 

i.e.:
TOC file 1 - TOC with REType Third Party Administrator for Product Plan A - ASO plans - INN Network ABC file 1 (REType Third Party Administrator)
TOC file 2 - TOC with REType Health Insurance Issuer for Product Plan A - Standard plans - INN Network ABC file 2 (REType Health Insurance Issuer)

Would it maybe make sense to remove the Reporting Entity Type from the INN files if we have a TOC file for our plan data? Since this data is already on the TOC file - does it make sense to keep it on the INN file if we are linking to those files in the TOC?

i.e. - if we didn't have to include the Reporting Entity Type attribute in INN when TOC is present:
TOC file 1 - TOC with REType Third Party Administrator for Product Plan A - ASO plans - INN Network ABC file 1 
TOC file 2 - TOC with REType Health Insurance Issuer for Product Plan A - Standard plans - INN Network ABC file 1 

Thoughts anyone??

**Comment by Devyandu** [comment]
@angelamarie0975 
I second the proposal. Th issue is something we have also been looking to solve.

If we are using the TOC file then having the reporting_entity_name and _type attributes in the INN file itself should not be required.
Having them there introduces an additional processing step of duplicating the files (provided by our issuers) and to populate the reporting entity. This moves us away from the desired design of sharing INN files. 

In the original proposal for [TOC ](https://github.com/CMSgov/price-transparency-guide/discussions/260) the intention was that we would remove the root node in INN files and start from the in_network arrray.

Thanks for raising this.

Tagging @shaselton-usds

**Comment by angelamarie0975** [comment]
@shaselton-usds Hi Scott - will you please provide some thought or an answer to the above questions? The same question is basically being asked in this discussion as well - https://github.com/CMSgov/price-transparency-guide/discussions/388

As of right now - we need some guidance on this so that we can proceed further to eliminate redundancy in files. Thank you.

**Comment by isaiah-r86** [comment]
We would also like to see this schema change because it would significantly cut down on data repetition for us.

@shaselton-usds could we get some guidance on this?

**Comment by shaselton-usds** [comment]
@angelamarie0975 @Devyandu @isaiah-r86 -- thanks for brining this up and the patience. 

This would certainly not be ideal to have duplicate files with only the [`reporting_entity_name` and `reporting_entity_type`](https://github.com/CMSgov/price-transparency-guide/blob/2df1618b6fe29f12cdcf55e81e5707da29c6608c/examples/allowed-amounts/allowed-amounts-multiple-plan-sample.json#L2-L3) being different. The reason they're there is to attach some identity to the files that are leveraging [table-of-contents files](https://github.com/CMSgov/price-transparency-guide/blob/2df1618b6fe29f12cdcf55e81e5707da29c6608c/examples/table-of-contents/table-of-contents-sample.json).

If those fields were made optional due to using table-of-contents files, there would only be negotiated rates (and minor meta data) included in the file. Would it make sense to add an attribute to the in-network files that would point back to the table-of-contents file location that refer to them? I'm not sure this is ideal. Thoughts?

**Reply by Devyandu** [reply]
Hi Scott,
The in-network files do not need to have any identity and any plan or TOC linkage. These are best kept as pure pricing contracts which were externalized since they were same across multiple plans.

Multiple plans nodes (perhaps even across multiple TOC files) could all reference the same in-network file. 

I think it would be good to make the reporting entity name and type fields optional in the schema and provide guidance that these are to be populated only if the TOC files are not being used. 

Thanks,
Devyandu

**Reply by Devyandu** [reply]
@shaselton-usds 
Hi Scott, Just checking if you have had a chance to look at this thread. It would be very helpful to have reporting entity name and type to be optional when a TOC file is being used.
Thanks

**Reply by angelamarie0975** [reply]
@shaselton-usds 
Yes - please have you had a chance to look into this some more? We really do need an answer to this as this kind of a blocker for us. We currently will have a work around and will just produce that same files twice - one as health insurance issuer and the other as third party administrator but we would really like to not have to have duplicate files if we don’t need to. Thank you!!

**Comment by isaiah-r86** [comment]
Hi @shaselton-usds , could you expand a little more on your suggestion?  Would the in-network-rate file contain an array of links to the table-of-contents files that leverage it?

I can't speak for the others in this discussion but our preference would be to make the reporting_entity_name and reporting_entity_type fields optional in the in-network-rates file if used with a table-of-contents.  Then link to those files from our table-of-contents files.

Is there a requirement to be able to link from the network rate files back to the ToCs?

**Comment by angelamarie0975** [comment]
@shaselton-usds Well - in our case, the file we are presenting to the public was going to be the TOC file which has the links to those negotiated rate files. Since the user would only be able to access the TOC files via the public website - meaning the other files are only displayed within that TOC, then shouldn't that be sufficient? If the user can't get to those network rate files outside of the TOC, then there wouldn't be a need to link them back to the TOC, would there? 

I know that our assumption and at least 1 of our associated organizations is presenting the TOC file in the same manner - I am thinking that with the implementation of the TOC files, that most organizations are probably assuming the same. 

I am in agreement with @isaiah-r86 - that the reporting_entity fields be made optional when there is a TOC file.

**Comment by angelamarie0975** [comment]
@shaselton-usds Just pinging this again to see if there has been any movement on it. Thank you!

**Comment by whiteck** [comment]
@shaselton-usds  have you had an opportunity to take a look at @isaiah-r86  recommendation?

**Comment by isaiah-r86** [comment]
@shaselton-usds now that these files have been out for a few months I'd like to bump this thread again.  Would it be possible to remove the reporting entity requirements in the in-network rate schema when used in conjunction with a table of contents?

**Comment by nmayle** [comment]
This discussion seems to assume that the report_entity on the in_network files is the same as the one in the table of contents file, but there are many examples out there of in_network files being produced by one entity and then used by another entity (see the various Blues). It is useful to know who produced the file and is therefore responsible for it.  It seems that the main objection to these fields is that when the in-network file is used by both a TPA and a fully-insured plan(s), then the reporting_entity_type would need to be different, but all of the other data would be the same.  

How about having a reporting_entity_type that describes an entity that acts as both a TPA and a health insurance issuer?  

There isn't yet a fixed list of reporting_entity_type values anyway, so adding one wouldn't even change the schema.
